api/v1alpha1/nodepool_types.go

```diff
@@ -211,3 +214,33 @@ type Filter struct {
 	// Values includes one or more filter values. Filter values are case-sensitive.
 	Values []string `json:"values"`
 }
+
+// Volume encapsulates the configuration options for the storage device
+type Volume struct {
```

Comment:
I'd suggest we consider whether the following are necessary to satisfy the use case:

* DeviceName
* Encrypted
* EncryptionKey

If not, we can narrow the API surface and testing matrix by removing them.

Reply:
Sounds good. They are removed now.